hack/gatewayapi-conformance.sh

```diff
@@ -32,22 +34,30 @@ cd "${CLONE_DIR}"
 # find the branch of gateway-api repo
 RELEASE_VERSION=$(\grep -oP "^\d+\.\d+" <<<"${BUNDLE_VERSION#v}")
 BRANCH="release-${RELEASE_VERSION}"
-echo "gateway-api repo branch: ${BRANCH}"
+echo "gateway-api repo branch \"${BRANCH}\" into ${CLONE_DIR}..."
 
 # clone the branch that matched the installed CRDs version
 # branch release-1.0 -> bundle-version v1.0.0
 # branch release-1.1 -> bundle-version v1.1.1
 # branch release-1.2 -> bundle-version v1.2.1
 git clone --branch "${BRANCH}" https://github.com/kubernetes-sigs/gateway-api
 cd gateway-api
+
+if [[ "$BUNDLE_VERSION" = "v1.2.1" ]]; then
+    echo "Cherry-picking fix for CoreDNS deployment issue in Gateway API v1.2.1"
```

Comment:
```suggestion
    // This is a temporary workaround until Gateway API v1.3.x.
    // See: https://github.com/kubernetes-sigs/gateway-api/pull/3389
    echo "Cherry-picking fix for CoreDNS deployment issue in Gateway API v1.2.1"
```

Reply:
If I push another change, I'll include this comment.  Otherwise, I'd rather not burn the CI cycles.  It's easy enough to look up the PR from the commit.